ckan/logic/action/get.py

```diff
@@ -1884,9 +1884,12 @@ def package_search(context, data_dict):
         include_drafts = asbool(data_dict.pop('include_drafts', False))
         data_dict.setdefault('fq', '')
         if not include_private:
-            data_dict['fq'] = '+capacity:public ' + data_dict['fq']
+            data_dict['fq'] = 'capacity:public' if not data_dict['fq'] \
```

Comment:
so, if `data_dict['fq']` is falsy, wouldn't it be safe to just use `'+capacity:public ' + data_dict['fq']` everytime?

Reply:
No, because the `+` between the two parts of the query is necessary. We also can't use `'+capacity:public +' + data_dict['fq']` every time, because if there's nothing in `data_dict[fq]`, this causes a dangling `+` which also breaks the query.